lib/gdpr/gateway/user_details.rb

```diff
@@ -79,12 +75,47 @@ def notify_inactive_users
       if user.mobile?
         WifiUser::SMSSender.send_credentials_expiring_notification(username, contact)
       elsif user.valid_email?(contact)
-        WifiUser::EmailSender.send_credentials_expiring_notification(username, contact)
+        begin
+          WifiUser::EmailSender.send_credentials_expiring_notification(username, contact)
+          @logger.info("Email sent to #{username} at #{contact}")
+        rescue Notifications::Client::BadRequestError => e
+          handle_email_error(e, username, contact)
+        rescue StandardError => e
+          @logger.warn(e.message)
+        end
       else
-        logger.warn("Invalid email address for user #{username}: #{contact}")
+        @logger.warn("Invalid contact for user #{username}")
       end
     end
 
-    logger.info("Finished notification process for users who were inactive for 11 months")
+    @logger.info("Finished notification process for users inactive for for 11 months")
+  end
+
+private
+
+  def find_inactive_users(days)
+    WifiUser::User.where(
+      Sequel.lit("
+      (last_login < DATE_SUB(NOW(), INTERVAL ? DAY)
+      OR (last_login IS NULL AND created_at < DATE_SUB(NOW(), INTERVAL ? DAY)))
+      AND username != 'HEALTH'", days, days),
+    ).all
+  end
+
+  def find_users_inactive_for_11_months(days)
```

Comment:
this name is a bit confusing. It looks for users that are inactive for exactly a to be defined number of days, not necessarily 11 months right?